Sources/StreamChatUI/ViewContainerBuilder/ViewContainerBuilder.swift

```diff
@@ -0,0 +1,270 @@
+    /// container.views {
+    ///     threadIconView
+    ///     replyTimestampLabel
+    /// }
+    /// ```
+    public static func buildBlock(_ components: UIView?...) -> [UIView] {
+        components.compactMap { $0 }
+    }
+
+    /// The block responsible to help creating additional constraints in a container.
+    /// Example:
+    /// ```
+    /// threadIconView.constraints {
+    ///   $0.heightAnchor.pin(equalToConstant: 15)
+    ///   $0.widthAnchor.pin(equalToConstant: 15)
+    /// }
+    /// ```
+    public static func buildBlock(_ components: NSLayoutConstraint...) -> [NSLayoutConstraint] {
+        NSLayoutConstraint.activate(components)
+        return components
+    }
+
+    /// A block responsible to support if-statements when building the stack views.
+    public static func buildEither(first component: UIStackView) -> UIStackView {
+        component
+    }
+
+    /// A block responsible to support if-statements when building the stack views.
+    public static func buildEither(second component: UIStackView) -> UIStackView {
+        component
+    }
+}
+
+/// The vertical container which represents a vertical `UIStackView`.
+///
+/// - parameter spacing: The spacing between views.
+/// - parameter distribution: The stack view distribution, by default it is `.fill`.
+/// - parameter alignment: The stack view alignment, by default it is `.fill`.
+/// - parameter content: The result builder responsible to return the stack view with the arranged views.
+public func VContainer(
+    spacing: CGFloat = 0,
+    distribution: UIStackView.Distribution = .fill,
+    alignment: UIStackView.Alignment = .fill,
+    @ViewContainerBuilder content: () -> UIStackView = { UIStackView() }
+) -> UIStackView {
+    let stack = content()
+    stack.translatesAutoresizingMaskIntoConstraints = false
+    stack.axis = .vertical
+    stack.distribution = distribution
+    stack.alignment = alignment
+    stack.spacing = spacing
+    return stack
+}
+
+/// The horizontal container which represents a horizontal `UIStackView`.
+///
+/// - parameter spacing: The spacing between views.
+/// - parameter distribution: The stack view distribution, by default it is `.fill`.
+/// - parameter alignment: The stack view alignment.
+/// - parameter content: The result builder responsible to return the stack view with the arranged views.
+public func HContainer(
+    spacing: CGFloat = 0,
+    distribution: UIStackView.Distribution = .fill,
+    alignment: UIStackView.Alignment = .fill,
+    @ViewContainerBuilder content: () -> UIStackView = { UIStackView() }
+) -> UIStackView {
+    let stack = content()
+    stack.translatesAutoresizingMaskIntoConstraints = false
+    stack.axis = .horizontal
+    stack.distribution = distribution
+    stack.alignment = alignment
+    stack.spacing = spacing
+    return stack
+}
+
+/// A flexible space that expands along the major axis of its containing stack layout.
+public func Spacer() -> UIView {
+    let view = UIView()
```

Comment:
Maybe here, we can use a UIstackview as it's a non rendering element, freeing some operations from happening on a spacer (eg drawing)

Reply:
I just wonder if creating an empty UIStackView might be a little more expensive than UIView 🤔 But I can change it to UIStackView for now, and we can see if there issues in the future

Reply:
Done ✅